### ISSUE TITLE
Some visualization method names and plot labels are misleading

### ISSUE THREAD

**Issue by zakv** [michaelhush/M-LOOP#85]
In particular:

* `GaussianProcessVisualizer.plot_hyperparameters_vs_run()`
  * This method actually plots fitted hyperparameter values as a function of fit number, not run number, so the name is a bit misleading. The x-axis label and docstring also say run number instead of fit number.
  * The title of the plot accurately says `'...vs fit number'`.
  * There is one fit per generation, so alternatively you could say it plots fitted hyperparameter values vs generation.
* `GaussianProcessLearner.plot_noise_level_vs_run()` has the same issues as `plot_hyperparameters_vs_run()`.
* `NeuralNetVisualizer.plot_losses()`
  * The loss is recorded [after every 10 epochs](https://github.com/michaelhush/M-LOOP/blob/bc172f251ef1fb71ca6c2df1f9d82f20f0eb7191/mloop/neuralnet.py#L279-L281).
  * The docstring, x-axis label, and title say its as a function of "training run", but generally there are more than 10 epochs per training run.

To make things more clear I was thinking of making the following changes:

* Rename the `GaussianProcessVisualizer` methods mentioned above to end in `vs_fit()`.
  * Update the x-axis labels and docstrings to say fit number.
  * Mention that there is one fit per generation in the docstring.
  * Keep the methods with the old "vs_run" names for now. Make them issue a deprecation warning then call the methods with the new names.
* Change the docstring, x-axis label, and title in `NeuralNetVisualizer.plot_losses()` to say epochs instead of training run.
  * Change the indices on the x-axis to be 0, 10, 20... instead of 0,1, 2...
  * Mention in the docstring that the loss is recorded every 10 epochs.

@charmasaur do you have any opinions on this? If not I'll just make the changes listed above, probably some time in the next week.

**Comment by charmasaur** [comment]
All sounds good to me. Sorry about the issues :(

**Comment by zakv** [comment]
Haha no worries, this one's an easy fix
